### ISSUE TITLE
Crash when MDE open and you subsequently connect to the computer via RDC

### ISSUE THREAD

**Issue by cyotek** [mike-ward/Markdown-Edit#224]
Hello,

I had MDE open on my main computer. As I sometimes do, I connected to this computer from another machine via RDC. The resolution of my main computer is 1920x1080. The resolution of the computer I used to RDC was 1680x1050. I always have the MDE window maximized.

I wasn't actually using MDE during my RDC so I don't know if this crash happened when I starting using RDC on the secondary computer, or when I resuming working on the min machine.

When I did resume on the machine machine, I noticed the following crash report was displayed on a popup message box

```
---------------------------

---------------------------
MarkdownEdit Crash Report

Date: 2017-01-07 09:59:26Z

Version: 1.32



-----------------------------------------------------------------

*** Pressing Ctrl+C will copy the contents of this dialog ***

*** Email to: mike@mike-ward.net ***

-----------------------------------------------------------------



Operation System Information

-----------------------------------------------------------------

Windows 10 Pro

Service Pack: No Service Pack detected





Exception

-----------------------------------------------------------------

System.ComponentModel.Win32Exception (0x80004005): The parameter is incorrect

   at Microsoft.Windows.Shell.WindowChromeWorker._HandleMoveForRealSize(WM uMsg, IntPtr wParam, IntPtr lParam, Boolean& handled) in d:\projects\git\MahApps.Metro\src\MahApps.Metro\MahApps.Metro.Shared\Microsoft.Windows.Shell\WindowChromeWorker.cs:line 1160

   at Microsoft.Windows.Shell.WindowChromeWorker._WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled) in d:\projects\git\MahApps.Metro\src\MahApps.Metro\MahApps.Metro.Shared\Microsoft.Windows.Shell\WindowChromeWorker.cs:line 612

   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)

   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)

   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)


---------------------------
OK   
---------------------------
```

I tested opening MDE and then remoting to the computer a few times, and can confirm that MDE crashes as soon I RDC into the computer.

The following crash log is also sometimes displayed when MDE tries to shut down after the first crash

```
---------------------------

---------------------------
MarkdownEdit Crash Report

Date: 2017-01-07 10:11:32Z

Version: 1.32



-----------------------------------------------------------------

*** Pressing Ctrl+C will copy the contents of this dialog ***

*** Email to: mike@mike-ward.net ***

-----------------------------------------------------------------



Operation System Information

-----------------------------------------------------------------

Windows 10 Pro

Service Pack: No Service Pack detected





Exception

-----------------------------------------------------------------

System.InvalidOperationException: Cannot set Visibility to Visible or call Show, ShowDialog, Close, or WindowInteropHelper.EnsureHandle while a Window is closing.

   at System.Windows.Window.VerifyNotClosing()

   at System.Windows.Window.InternalClose(Boolean shutdown, Boolean ignoreCancel)

   at System.Windows.Window.Close()

   at MarkdownEdit.Commands.CloseCommand.Execute(Object sender, ExecutedRoutedEventArgs e) in C:\Users\mike\GitHub\Markdown-Edit\src\MarkdownEdit\Commands\CloseCommand.cs:line 18

   at System.Windows.Input.CommandBinding.OnExecuted(Object sender, ExecutedRoutedEventArgs e)

   at System.Windows.Input.CommandManager.ExecuteCommandBinding(Object sender, ExecutedRoutedEventArgs e, CommandBinding commandBinding)

   at System.Windows.Input.CommandManager.FindCommandBinding(CommandBindingCollection commandBindings, Object sender, RoutedEventArgs e, ICommand command, Boolean execute)

   at System.Windows.Input.CommandManager.FindCommandBinding(Object sender, RoutedEventArgs e, ICommand command, Boolean execute)

   at System.Windows.Input.CommandManager.OnExecuted(Object sender, ExecutedRoutedEventArgs e)

   at System.Windows.UIElement.OnExecutedThunk(Object sender, ExecutedRoutedEventArgs e)

   at System.Windows.Input.ExecutedRoutedEventArgs.InvokeEventHandler(Delegate genericHandler, Object target)

   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)

   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)

   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)

   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)

   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)

   at System.Windows.Input.RoutedCommand.ExecuteImpl(Object parameter, IInputElement target, Boolean userInitiated)

   at System.Windows.Input.RoutedCommand.Execute(Object parameter, IInputElement target)

   at MarkdownEdit.Controls.Preview.<>c.<.ctor>b__8_3(Object s, RoutedEventArgs e) in C:\Users\mike\GitHub\Markdown-Edit\src\MarkdownEdit\Controls\Preview.xaml.cs:line 60

   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)

   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)

   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)

   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)

   at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)

   at System.Windows.BroadcastEventHelper.BroadcastUnloadedEvent(Object root)

   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)


---------------------------
OK   
---------------------------
```

Usually followed by two of Windows's Application Error dialogs which would write these entries (Faulting process ID, Faulting application start time and Report ID were unique to each entry, the other values remain the same)

```
Faulting application name: mde.exe, version: 1.32.0.0, time stamp: 0x586a9f1c
Faulting module name: KERNELBASE.dll, version: 10.0.14393.479, time stamp: 0x58256d37
Exception code: 0xc0020001
Fault offset: 0x000da832
Faulting process ID: 0x88744
Faulting application start time: 0x01d268ce566d3d1c
Faulting application path: C:\Users\Richard\AppData\Local\MarkdownEdit\bin\mde.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report ID: 641cdae5-1bf2-4f8d-8ed3-accd626f8269
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by punker76** [comment]
@cyotek Thx for the stack traces and step by step explained reproduction. It's the MahApps lib which causes this issue. I'll investigate now a little bit more. It seems it has something to do with the remote stuff.

**Comment by cyotek** [comment]
Hello,

I can't help but feel there is more than one issue involved with this. As you mention MahApps you can interchange MDE with MahApps in the rest of this reply as appropriate.

I did some most tests, and have the following observations

* Changing the resolution of the main machine to match the RDC client didn't help
* I suspect it's not the fact that MDE was maximized, but that the "normal" window state was taller than the new resolution (if I shrank the normal window state to be smaller than 1680x1050 then that would get rid of the first crash in my original issue)
* No matter what, as soon as I either RDC into the computer, or directly resume my session on the computer, MDE immediately closes. It doesn't seem to crash (no exception logs or event entries), it just closes. This would seem to account for the second trace I posted as it tried to shutdown after the first crash occurred.

So I think there are two issues:
1. MDE crashes if the screen resolution changes, and the normal MDE window state is larger on any axis than the new resolution. However, this only seems to apply to the RDC resolution change. If I try to simulate by just changing the resolution on the main machine, it is gracefully handled and the window resized to fit
2. MDE seems to execute a normal shutdown when you either RDC to the computer, or resume a direct connection. I tested both scenarios, but why it's doing that is a complete mystery - the first bug at least makes sense, this one doesn't :)

Hope this helps, and makes sense - if not let me know. Happy to run additional tests if need be.

Regards;
Richard Moss

**Comment by mike-ward** [comment]
@punker76  Thanks for looking into this.

**Comment by punker76** [comment]
@mike-ward Playing with latest source and figured out that the browser fires the Unloaded event 😯 at the `Preview`

- open remote desktop connection and login
- start mde
- close rdc
- open rdc again and login with same user
- wuup mde closes from this unloaded event

```csharp
            Browser.Unloaded += (s, e) =>
            {
                try
                {
                    ApplicationCommands.Close.Execute(null, Application.Current.MainWindow);
                }
                catch (NullReferenceException)
                {
                }
            };
```

**Comment by mike-ward** [comment]
So opening an RDC fires unloaded event in the browser. Odd indeed. I'm trying to remember why that's in there in the first place.

**Comment by cyotek** [comment]
This doesn't happen in the latest version; I assume because of cb62e420220880b163ccb7b2eb4a46b11586ed69. If that's the case then this issue can probably be closed (along with #239, I briefly tested that too)

**Comment by mike-ward** [comment]
Thanks for the follow up
